src/router/index.ts

```diff
@@ -1,10 +1,16 @@
 import { createRouter, createWebHistory } from "vue-router";
 
+const requiredAuthKeys = ["userId", "apiKey"];
+
 const routes = [
   {
     path: "/quiz/:quizId",
     name: "Player",
-    props: true,
+    props: (route: any) => ({
+      quizId: route.params.quizId,
```

Comment:
Instead of explicitly specifying `quizId` ka param, you can just do this
```js
 props: (route: any) => ({
      ...route.params,
      thirdPartyUserId: route.query.userId,
      thirdPartyApiKey: route.query.apiKey,
    }),
```